### ISSUE TITLE
Add extra field to Cancel API indicating reason of cancellation

### ISSUE THREAD

**Issue by isabelcosta** [anitab-org/mentorship-backend#66]
## Description
As a user,
I need to explain mentorship relation cancellation reason,
so that I can help improving the system with proper feedback.

### Update
- [ ] Optional text field, e.g.: `cancellation_reason`, to API request data of Cancel API and Mentorship relation database model
- [ ] Add and fix tests for this
- [ ] Document properly on Swagger UI

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
3 hours

**Comment by fnshemerirwe** [comment]
@isabelcosta, can I please work on this issue?

**Comment by isabelcosta** [comment]
Sure @flacode :) go ahead!

**Comment by bhawesh96** [comment]
If this is stale, I'd like to work on it!

**Comment by isabelcosta** [comment]
@flacode are you still working on this?

**Comment by isabelcosta** [comment]
@bhawesh96 you can work on this if you want, since there are no updates :) Do you still want to work on this?

**Comment by cribblportal** [comment]
Yes, I'll take it up

**Comment by bhawesh96** [comment]
Yes, I'll take it up

**Comment by bhawesh96** [comment]
Can we store the reason for cancellation under `notes` (overwriting the existing value) or do I make a new column?

**Comment by bhawesh96** [comment]
Since `cancellation_reason` has to be sent in the request, will we need to change the request type from PUT to POST?

**Comment by isabelcosta** [comment]
@bhawesh96
> Can we store the reason for cancellation under notes (overwriting the existing value) or do I make a new column?

 I think that overriding `notes` isn't a great option, because we may still want to know the details of the relationship even after cancelation, for historical purposes. So its best to create the new field in the `MentorshipRelationModel` class.

> Since cancellation_reason has to be sent in the request, will we need to change the request type from PUT to POST?

I think it should remain PUT, you won't be creating anything, just updating the relationship with the state from `ACCEPTED` to `CANCELLED` and updating the `cancellation_reason` field.

**Comment by sharifa2708** [comment]
@isabelcosta I would like to work on this issue

**Comment by isabelcosta** [comment]
I will review your PR first, so please do the changes and then work on this. @sharifa2708

**Comment by sharifa2708** [comment]
@isabelcosta Sure I will make the changes suggested

**Comment by CleverFool77** [comment]
Hi @isabelcosta Is this issue available?

**Comment by isabelcosta** [comment]
Yes it is :) @CleverFool77 you can work on it if you want. Do you?

**Comment by CleverFool77** [comment]
Yes I would like to take this up.
Thanks @isabelcosta

**Comment by sumit-badsara** [comment]
@isabelcosta I want to work on this issue, if it's available.

**Comment by isabelcosta** [comment]
@BadduCoder I think you can work on this yes since we have no recent updates on this issue.
cc @CleverFool77

**Comment by isabelcosta** [comment]
@BadduCoder I left you a note on your latest PR, please address that before moving onto this issue ;)

**Comment by sumit-badsara** [comment]
@isabelcosta Shouldn't we consider who(Mentor/Mentee) cancelled the relation, or just update the reason without taking in consideration who did it.

**Comment by sumit-badsara** [comment]
@isabelcosta I will start working on this once the previous PR is approved. Till then I will explore the project where the changes are to be done and discuss doubts if I encounter any.

**Comment by sumit-badsara** [comment]
@isabelcosta I tried updating swagger.json, but it doesn't reflects changes on Swagger UI. Any suggestions where I might be doing something wrong?

**Comment by sumit-badsara** [comment]
@isabelcosta Updated the PR! 😄
